### PR TITLE
test(ui): rejigger the collector tests to be less flaky

### PR DESCRIPTION
**Pull request by hoorayimhelping**
Makes collector e2e test lest flaky

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: ui/cypress/e2e/collectors.test.ts [hunk 1]
```diff
@@ -138,52 +147,76 @@ describe('Collectors', () => {
...
+                            )
+                          })
+                      })
+                  })
+              })
+          })
 
-        bucketz.sort()
+        // sort by buckets test here
+        cy.reload()
```
Comment (121watts) [inline ui/cypress/e2e/collectors.test.ts]: why reload here?

File: ui/cypress/e2e/collectors.test.ts [hunk 2]
```diff
@@ -196,9 +229,9 @@ describe('Collectors', () => {
               expect(text).to.include(bucketz[index])
             })
           })
-      })
-      // sort by name test here
-      it('can sort telegraf configs by name', () => {
+
+        // sort by name test here
+        cy.reload()
         cy.getByTestID('collector-card--name').should('have.length', 3)
```
Comment (121watts) [inline ui/cypress/e2e/collectors.test.ts]: and here?

File: ui/cypress/e2e/collectors.test.ts [hunk 3]
```diff
@@ -21,30 +21,41 @@ describe('Collectors', () => {
...
-        cy.getByTestID('streaming').within(() => {
-          cy.get('.cf-button')
-            .contains('Listen for Data')
+      cy.getByTestID('table-row')
+        .should('have.length', 0)
+        .then(() => {
+          cy.contains('Create Configuration')
             .click()
+            .then(() => {
+              cy.getByTestID('overlay--container').then(() => {
```
Comment (121watts) [inline ui/cypress/e2e/collectors.test.ts]: `.within` achieves the same sequential behavior as a `then` but also guarantees the rest of our selectors will be within said container.  
```suggestion
              cy.getByTestID('overlay--container').within(() => {
```

File: ui/cypress/e2e/collectors.test.ts (hunk 1 above)
Comment (hoorayimhelping) [reply ui/cypress/e2e/collectors.test.ts]: it's more a conceptual reload than anything. just kind of like a break in the last bit of tests. i'll remove them

File: ui/cypress/e2e/collectors.test.ts (hunk 3 above)
Comment (hoorayimhelping) [reply ui/cypress/e2e/collectors.test.ts]: mistakenly removed this one, didn't go back through the undo stack far enough i think heh

File: ui/cypress/e2e/collectors.test.ts (hunk 1 above)
Comment (hoorayimhelping) [reply ui/cypress/e2e/collectors.test.ts]: so i smooshed three separate specs into a single spec to limit teardown and rebuilding and such, but a consequence of that these is that there is some wonky leftover state from the previous test that affects the current test (like active filtering or sorting for example). these `reload`s are necessary to clear out that state

**Pull request merged**